### PR TITLE
Update asix-ax88179 from 2.16.0 to 2.17.0

### PR DESCRIPTION
**Pull request by coreygo**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

### CODE REVIEW

**Comment by coreygo** [comment]
Updates `uninstall early_script` to v2.16.0.

File: Casks/asix-ax88179.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@
   end
 
   uninstall early_script: {
-                            executable: "#{staged_path}/AX88179_178A_Uninstall_v150.command",
+                            executable: "#{staged_path}/AX88179_178A_Uninstall_v1.6.0.command",
```
Comment (miccal) [inline Casks/asix-ax88179.rb:30]: Is this correct for both version 2.14.0 and 2.16.0?

Comment (coreygo) [reply Casks/asix-ax88179.rb:30]: `AX88179_178A_Uninstall_v150.command` is still used for v2.14. If else statement needed for MacOS.version around uninstall?

Comment (miccal) [reply Casks/asix-ax88179.rb:30]: Yes I think so, thanks.

Comment (coreygo) [reply Casks/asix-ax88179.rb:30]: Looks like v2.17.0 has been released now as well, so I'll update that too.

Can you point me to an example uninstall stanza with an if else statement? I'm honestly not all that familiar with the formula syntax and don't want to make an assumption. I checked the [uninstall stanza info](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/uninstall.md) for details but I'd like to get it right before resubmitting for review. 🖖

Comment (miccal) [reply Casks/asix-ax88179.rb:30]: @coreygo like this:
```
  if MacOS.version <= :mojave
    uninstall early_script: {
                              executable: "#{staged_path}/AX88179_178A_Uninstall_v150.command",
                            },
              kext:         'com.asix.driver.ax88179-178a',
              pkgutil:      'com.asix.pkg.ax88179-178a*'
  else
    uninstall early_script: {
                              executable: "#{staged_path}/AX88179_178A_Uninstall_v1.6.0.command",
                            },
              kext:         'com.asix.driver.ax88179-178a',
              pkgutil:      'com.asix.pkg.ax88179-178a*'
  end
```

Comment (coreygo) [reply Casks/asix-ax88179.rb:30]: Sweet, thanks, I wasn't sure if I needed to duplicate kext and pkgutil.

Comment (coreygo) [reply Casks/asix-ax88179.rb:30]: @miccal looking at the ASIX website again, was about to add the if and update to v2.17.0 to close this out.

It appears they're specifying v2.17.0 applies to `<= :mojave` now ["Apple Mac OS X 10.9 to 10.15 Drivers Installer | v2.17.0 | 2020-06-19"](https://www.asix.com.tw/download.php?sub=driverdetail&PItemID=131).

So I'm wondering if we need multiple versions anymore—ASIX doesn't list previous MacOS versions.

Comment (miccal) [reply Casks/asix-ax88179.rb:30]: @coreygo in that case, I would suggest you remove the `if/else` statements and just make this for the one version.

**Comment by coreygo** [comment]
Install succeeded. Tested locally.

<details><summary>CI build log</summary>

```
2020-07-11T00:34:36.2447790Z installer: The install was successful.
2020-07-11T00:34:36.2448650Z installer: The install requires restarting now.
2020-07-11T00:34:36.9363090Z com.asix.driver.ax88179-178a failed to load - (libkern/kext) system policy prevents loading; check the system/kernel logs for errors or try kextutil(8).
2020-07-11T00:34:36.9378330Z [34m==>[0m [1mPurging files for version 2.17.0 of Cask asix-ax88179[0m
2020-07-11T00:34:36.9598940Z [31mError:[0m Failure while executing; `/usr/bin/sudo -E -- /sbin/kextload -b com.asix.driver.ax88179-178a` exited with 27. Here's the output:
2020-07-11T00:34:36.9600020Z [31mcom.asix.driver.ax88179-178a failed to load - (libkern/kext) system policy prevents loading; check the system/kernel logs for errors or try kextutil(8).
2020-07-11T00:34:36.9600740Z [0m
```

</details>

It looks like uninstall needs some updating still due to ASIX dmg/pkg changes.

Assuming `AX88179_178A_Uninstall_v1.7.0.pkg` needs to be specified as `pkgutil` not `early_script`now?

```
$ "$(brew --repository)/Library/Taps/homebrew/homebrew-cask/developer/bin/list_recent_pkg_ids"
com.asix.pkg.ax88179-178a-10.15
com.asix.ax88179.uninstall
```

Packages appear to have changed too.

**Comment by coreygo** [comment]
<details><summary>Contents of `AX88179_178A_Uninstall_v1.6.0.command` in `AX88179_178A_Uninstall_v1.7.0.pkg` for reference…</summary>

```
$ cat AX88179_178A_Uninstall_v1.6.0.command
#!/bin/sh

KEXT_PATH_ETH_108_179=/System/Library/Extensions/IONetworkingFamily.kext/Contents/PlugIns/AX88179.kext
KEXT_PATH_ETH_108=/System/Library/Extensions/IONetworkingFamily.kext/Contents/PlugIns/AX88179_178A.kext
KEXT_PATH_ETH_109=/Library/Extensions/AX88179_178A.kext

echo "AX88179_178A_Uninstall_v1.6.0"

if [ -e $KEXT_PATH_ETH_108_179 ]; then
	sudo kextunload "$KEXT_PATH_ETH_108_179"
	sudo rm -rf "$KEXT_PATH_ETH_108_179"
fi

if [ -e $KEXT_PATH_ETH_108 ]; then
	sudo kextunload "$KEXT_PATH_ETH_108"
	sudo rm -rf "$KEXT_PATH_ETH_108"
fi

if [ -e $KEXT_PATH_ETH_109 ]; then
	sudo kextunload "$KEXT_PATH_ETH_109"
	sudo rm -rf "$KEXT_PATH_ETH_109"
fi

pkgutil --pkgs| grep 'AX88179.pkg' &> /dev/null
if [ $? == 0 ]; then
   sudo pkgutil --forget AX88179.pkg
fi

pkgutil --pkgs| grep 'AX88179_178A.pkg' &> /dev/null
if [ $? == 0 ]; then
   sudo pkgutil --forget AX88179_178A.pkg
fi

pkgutil --pkgs| grep 'com.asix.tw.ax88179_178a' &> /dev/null
if [ $? == 0 ]; then
   sudo pkgutil --forget com.asix.tw.ax88179_178a
fi

pkgutil --pkgs| grep 'com.asix.pkg.ax88179-178a-10.8' &> /dev/null
if [ $? == 0 ]; then
   sudo pkgutil --forget com.asix.pkg.ax88179-178a-10.8
fi

pkgutil --pkgs| grep 'com.asix.pkg.ax88179-178a-10.9' &> /dev/null
if [ $? == 0 ]; then
   sudo pkgutil --forget com.asix.pkg.ax88179-178a-10.9
fi

pkgutil --pkgs| grep 'com.asix.pkg.ax88179-178a-10.15' &> /dev/null
if [ $? == 0 ]; then
   sudo pkgutil --forget com.asix.pkg.ax88179-178a-10.15
fi



sudo kextcache --clear-staging
sudo touch /System/Library/Extensions/
sudo touch /Library/Extensions/
sudo kextcache -i / &> /dev/null

echo "Uninstall Complete."
```

</details>

**Comment by miccal** [comment]
It seems to me that the `AX88179_178A_Uninstall_v1.7.0.pkg` takes care of everything, so perhaps just this is necessary for the `uninstall`:
```
  uninstall script: {
                      executable: "#{staged_path}/AX88179_178A_Uninstall_v1.7.0.pkg",
                    }
```

**Comment by coreygo** [comment]
@miccal I thought that might be the case too, apparently not.

<details><summary>Uninstall script `executable: "#{staged_path}/AX88179_178A_Uninstall_v1.7.0.pkg"` with v2.17.0</summary>

```
$ brew cask uninstall asix-ax88179
==> Uninstalling Cask asix-ax88179
==> Running uninstall script /usr/local/Caskroom/asix-ax88179/2.17.0/AX88179_178A_Uninstall_v1.7.0.pkg
/usr/local/Caskroom/asix-ax88179/2.17.0/AX88179_178A_Uninstall_v1.7.0.pkg: /usr/local/Caskroom/asix-ax88179/2.17.0/AX88179_178A_Uninstall_v1.7.0.pkg: cannot execute binary file
Error: Failure while executing; `/usr/local/Caskroom/asix-ax88179/2.17.0/AX88179_178A_Uninstall_v1.7.0.pkg` exited with 126. Here's the output:
/usr/local/Caskroom/asix-ax88179/2.17.0/AX88179_178A_Uninstall_v1.7.0.pkg: /usr/local/Caskroom/asix-ax88179/2.17.0/AX88179_178A_Uninstall_v1.7.0.pkg: cannot execute binary file

Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
/usr/local/Homebrew/Library/Homebrew/system_command.rb:176:in `assert_success!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:53:in `run!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:29:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/abstract_uninstall.rb:297:in `uninstall_script'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/abstract_uninstall.rb:79:in `uninstall_early_script'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/abstract_uninstall.rb:68:in `dispatch_uninstall_directive'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/uninstall.rb:11:in `block in uninstall_phase'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/uninstall.rb:10:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/uninstall.rb:10:in `uninstall_phase'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:455:in `block in uninstall_artifacts'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/set.rb:777:in `each'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/set.rb:777:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:451:in `uninstall_artifacts'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:400:in `uninstall'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/uninstall.rb:24:in `block in run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/uninstall.rb:14:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/uninstall.rb:14:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/abstract_command.rb:36:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:150:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:92:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:9:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:111:in `<main>'
```

</details>

I also tried `"#{staged_path}/AX88179_178A_Uninstall_v1.7.0.pkg/AX88179_178A_Uninstall_v1.6.0.command"` knowing in advance it wouldn't work, it didn't. 🤓

**Comment by miccal** [comment]
Ah crap, I should have paid more attention to what you had changed (a script to a package) -- replace the `uninstall` stanza with the following:
```
  uninstall_preflight do
    system_command '/usr/sbin/installer',
                   args: [
                           '-pkg', "#{staged_path}/AX88179_178A_Uninstall_v1.7.0.pkg",
                           '-target', '/'
                         ],
                   sudo: true
  end
``` 
and let's see how the tests go.

**Comment by coreygo** [comment]
@miccal tested install and uninstall locally, that seems to do the trick, thanks for the help.

Cask audit complained about completely removing the `uninstall` stanza because of `pkg`:
```
• audit for asix-ax88179: warning
 - installer and pkg stanzas require an uninstall stanza
```
Removed `early_script`, left the rest.

CI install fails but that seems expected with `asix-ax88179` CI checks looking at past pull requests.

**Comment by vitorgalvao** [comment]
Verified as working on a VM.

**Comment by coreygo** [comment]
@vitorgalvao No merge? Should I create a fresh pull request?

**Comment by notpeter** [comment]
I tested this and it worked for me. @vitorgalvao did you intend to close this without merging?

**Comment by vitorgalvao** [comment]
I intended to merge, yes. Will do so when I get back to the Mac, because PRs with failing tests cannot be merged from a phone.

**Pull request merged**